src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/CumulativeFilter.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature;
+
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.exception.SolverException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Abstract class for cumulative constraint filtering algorithms.
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public abstract class CumulativeFilter {
+    protected Task[] tasks;
+    protected IntVar[] heights;
+    protected IntVar capacity;
+    protected Propagator<IntVar> aCause;
+
+    protected ArrayList<Integer> indexes;
```

Comment:
Use `TIntList` instead

Reply:
TIntList can not do specific sorting (typically this kind of sorting :  `indexes.sort(Comparator.comparingInt(j -> -tasks[j].getEnd().getUB()));`), which is why I used ArrayList for this purpose.